### DISCUSSION TITLE
I can't get token with getToken({req})

### DISCUSSION THREAD

**Discussion by ekofi** [nextauthjs/next-auth#4129] [Help] [answered]
### Question 💬

Hi, I can't get the token with getToken:

![aaan](https://user-images.githubusercontent.com/30902468/156883361-2770e73a-0ac4-4acd-8efa-02b2b815a981.png)

This variables are ok:
NEXTAUTH_SECRET=secret
NEXTAUTH_URL=http://localhost:3000

### How to reproduce ☕️

[...nextauth].js
```
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";

export default NextAuth({
  // Configure one or more authentication providers
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
      authorizationUrl:
        "https://accounts.google.com/o/oauth2/v2/auth?prompt=consent&access_type=offline&response_type=code",
      scope:
        "https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/analytics.readonly",
    }),
    // ...add more providers here
  ],
  jwt: {
    secret: process.env.JWT_SECRET,
    encryption: true,
  },
  secret: process.env.NEXTAUTH_SECRET,
  callbacks: {
    async redirect({ url, baseUrl }) {
      return Promise.resolve(url);
    },
    async jwt({ token, user, account, profile, isNewUser }) {
      return token;
    },
    async session({ session, user, token }) {
      return session;
    },
  },
});

```
API section:

```
import { getToken } from "next-auth/jwt";

const secret = process.env.NEXTAUTH_SECRET;

export default async (req, res) => {
  const token = await getToken({ req, secret, encryption: true });
  console.log(token);
  if (token) {
    // Signed in
    console.log("JSON Web Token", JSON.stringify(token, null, 2));
  } else {
    // Not Signed in
    res.status(401);
  }
  res.end();
};
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment, marked as answer]
From the error stack, it looks like you are trying to use an API Route handler on a non-API Route (pages/dashboard).

Please see: https://nextjs.org/docs/api-routes/introduction

**Reply by ekofi** [reply]
But I just called the API with useEffect, nothing more. ```
useEffect(() => {
    getAnalyticsData();
  }, []);
```

`And my API`

```
import { getToken } from "next-auth/jwt";

const secret = process.env.NEXTAUTH_SECRET;

export default async (req, res) => {
  /* const token = await getToken({ req, secret, encryption: true }); */
  /* if (token) {
    // Signed in
    console.log("JSON Web Token", JSON.stringify(token, null, 2));
  } else {
    // Not Signed in
    res.status(401);
  }
  res.end(); */
  /* const session = await getSession({ req });
  const token = await getToken({ req });

  console.log(token);

  if (!session) {
    return res.status(401).end();
  }

  const data = await fetch(
    "https://www.googleapis.com/analytics/v3/management/accountSummaries",
    {
      headers: {
        Authorization: `Bearer ${token.access_token}`,
      },
    }
  )
    .then((response) => response.json())
    .then((data) => console.log(data));

  res.status(200).json(data); */
};
```
